attempt_handler_design.md

```diff
@@ -0,0 +1,143 @@
+# Design Proposal: The `AttemptHandler` Service for Reliable Payment Resolution
+
+**Author:** ziggie
+**Date:** August 7, 2025
+**Status:** Proposed
+
+## 1. Problem Statement: Stuck In-Flight Payments
+
+The current payment architecture in `lnd` suffers from a critical flaw where payments can become permanently stuck in the `INFLIGHT` state.
+
+### Root Cause
+
+The root cause is the **tight coupling** between the `paymentLifecycle` (the process that sends a payment) and the `resultCollector` (the goroutine that waits for the payment's result). The `resultCollector`'s lifetime is tied directly to the `paymentLifecycle`.
+
+When the `paymentLifecycle` exits for any reason other than definitive success or failure—such as a context timeout, user cancellation, or a route-finding error—it closes a `quit` channel. This action immediately terminates all of its associated `resultCollector` goroutines.
```

Comment:
I'm not sure this analysis is correct - if the payment has pending htlcs and timed out, the payment is still considered as inflight and will wait for attempt results? Think it's handled in `decideNextStep`.

Reply:
I think if we really want to make "sending a payment" non-blocking, I think there is no other way than separating the attempt collection, otherwise we will not be able to exit a payment after the timeout hit.